src/components/analytics/index.ts

```diff
@@ -23,5 +45,44 @@ if (typeof GOOGLE_ANALYTICS_ID === 'string') {
       : 'mobileRegular',
   })
 } else {
-  ReactGA.initialize('test', { gtagOptions: { debug_mode: true } })
+  googleAnalytics.initialize('test', { gtagOptions: { debug_mode: true } })
+}
+
+function reportWebVitals({ name, delta, id }: Metric) {
+  gaCommandSendTiming('Web Vitals', name, Math.round(name === 'CLS' ? delta * 1000 : delta), id)
 }
+
+// tracks web vitals and pageviews
+export function GoogleAnalyticsReporter({ location: { pathname, search } }: RouteComponentProps): null {
```

Comment:
This is not presentational, it just runs hooks. Instead of adding it as a component, make it a hook that is run on the root component.

Reply:
Sounds good done.